exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/compression/DrillCompressionCodecFactory.java

```diff
@@ -86,6 +99,9 @@ public BytesInputCompressor getCompressor(CompressionCodecName codecName) {
           codecName,
           c -> new AirliftBytesInputCompressor(codecName, allocator)
       );
+    } else if (codecName == CompressionCodecName.GZIP) {
+      // hack for gzip: construct a new codec factory every time to avoid a concurrrency bug c.f. DRILL-8139
+      return CodecFactory.createDirectCodecFactory(config, allocator, pageSize).getCompressor(codecName);
```

Comment:
If I recall correctly, the codec factory should manage the full lifecycle of compressors / decompressors, release created codecs when calling its `CompressionCodecFactory.release()` method, and so on. But with this change, it will be impossible to do that.